rotkehlchen/db/filtering.py

```diff
@@ -84,23 +84,33 @@ def prepare(self) -> Tuple[List[str], List[Any]]:
 
 
 @dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
-class DBETHTransactionAddressFilter(DBFilter):
+class DBETHTransactionJoinsFilter(DBFilter):
     """Find transactions involving any of the addresses. Including in an internal.
 
     This uses the mappings we create in the DB at transaction addition to signify
     relevant addresses for a transaction.
     """
-    addresses: List[ChecksumEthAddress]
+    addresses: Optional[List[ChecksumEthAddress]]
+    join_events: bool = False
 
     def prepare(self) -> Tuple[List[str], List[Any]]:
-        questionmarks = '?' * len(self.addresses)
-        filters = [
-            f'AS A LEFT OUTER JOIN ethtx_address_mappings AS B WHERE '
-            f'A.tx_hash=b.tx_hash AND B.address IN ({",".join(questionmarks)})',
-        ]
-        bindings = self.addresses
+        query_filter, bindings = '', []
+        if self.join_events:
+            query_filter += (
+                ' INNER JOIN '
+                '(SELECT event_identifier, counterparty, asset FROM history_events)'
+                'ON hex(ethereum_transactions.tx_hash)=substr(event_identifier, 3)'
+            )
+        if self.addresses:
```

Comment:
explicit checks of what is a truthy value

```suggestion
        if self.addresses is not None:
```

Reply:
But what if addresses is `[]`? In this case the query won't return anything. Is it okay?

Reply:
Well I would say an empty addresses list to filter should never make it down here. We should prevent it at api level.

But semantically if you tell me to filter for a list of addresses and that list is empty the query should indeed return nothing

Reply:
Alright!